### DISCUSSION TITLE
Docs outdated?

### DISCUSSION THREAD

**Discussion by oleksii-lukin** [gatsbyjs/gatsby#30652] [Help] [answered]
https://www.gatsbyjs.com/plugins/gatsby-plugin-image

Docs saying query should be like this:

```graphql
query {
 blogPost(id: { eq: $Id }) {
   title
   body
   avatar {
     childImageSharp {
       gatsbyImageData(width: 200)
     }
   }
 }
}
```

but that was throwing me an error `Cannot query field "gatsbyImageData" on type "ImageSharp".`

while what worked for me looks like this

```graphql
{
  file(relativePath: {eq: "shared/glasses-image.png"}) {
    childImageSharp {
      gatsbyImage(layout: FLUID, maxWidth: 664) {
        imageData
      }
    }
  }
}
```

Am I using something different, or docs have outdated information?

**Comment by LekoArts** [comment, marked as answer]
The two example queries you posted here are two completely separate things. The first one assumes that you have a `blogPost` type with an `avatar` pointing to an image. If you don't have that in your local project then it won't work. It's an example explaining the query, not necessarily good to copy/paste into an existing project.

**Reply by oleksii-lukin** [reply]
I think the reason was that I was using gatsby 2 and gatsby-plugin-image is not compatible with it. After updating gatsby and all plugins to version 3, everything started working as described. Got some discussion with more details here https://github.com/gatsbyjs/gatsby/discussions/30657

**Reply by oleksii-lukin** [reply]
@LekoArts btw since gatsby using monorepo, how do I search information for particular plugin? I tried to look for contentful hoping to find anything related to contentful inside gatsby-plugin-image. Found half thousand (or so) things all over gatsby repo without ability to filter ones related to gatsby-plugin-image. Doesn't look like github supports something like this. Is asking questions here only option? Or I can use some "tags" or keywords to search information only related to particular plugin?

**Reply by LekoArts** [reply]
You can use the plugin search :)

https://www.gatsbyjs.com/plugins/gatsby-source-contentful => https://www.gatsbyjs.com/plugins/gatsby-source-contentful#using-the-new-gatsby-image-plugin
https://www.gatsbyjs.com/plugins/gatsby-plugin-image

**Reply by oleksii-lukin** [reply]
I doubt there is any useful information there. What I meant is search for issues which similar to mine. How I can do it here?

![image](https://user-images.githubusercontent.com/1281970/113709370-15134f00-96eb-11eb-853d-fd3b4173ddee.png)

Is attaching `gatsby-plugin-image` keyword a safe option?

**Reply by oleksii-lukin** [reply]
In the end I think that was how I fixed my problem - looked for similar issues all over github

**Reply by LekoArts** [reply]
You could use that label: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22topic%3A+images%2Fmedia%22

**Reply by oleksii-lukin** [reply]
Aha, that's how I do it! Thank you.
